error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ComparatorRules.java

```diff
@@ -243,18 +245,77 @@ int after(T value1, T value2) {
     }
   }
 
+  /** Prefer {@link Collections#sort(List)} over more verbose alternatives. */
+  static final class SortCollections<T extends Comparable<? super T>> {
+    @BeforeTemplate
+    void before(List<T> collection) {
+      Collections.sort(collection, naturalOrder());
+    }
+
+    @AfterTemplate
+    void after(List<T> collection) {
+      Collections.sort(collection);
+    }
+  }
+
+  /**
+   * Avoid unnecessary creation of a {@link Stream} to determine the minimum of a known collection
+   * of values.
+   */
+  static final class MinOfArray<S, T extends S> {
+    @BeforeTemplate
+    T before(T[] array, Comparator<S> cmp) {
+      return Arrays.stream(array).min(cmp).orElseThrow();
+    }
+
+    @AfterTemplate
+    T after(T[] array, Comparator<S> cmp) {
+      return Collections.min(Arrays.asList(array), cmp);
+    }
+  }
+
+  /** Prefer {@link Collections#min(Collection)} over more verbose alternatives. */
+  static final class MinOfCollection<T extends Comparable<? super T>> {
```

Comment:
I see how this matches the other rule (should have thought this through better in the other PR :grimacing:), but I realize this one should be named `CollectionsMin`, and similarly before for the "max" rule. I'll also move it up a bit and rename the `MinOfCollectionComparator` variant to `CollectionsMinWithComparator`; the "with" naming is also used elsewhere. (Not sure this'll be the final naming we'll settle on :thinking:.)